src/main/java/com/squareup/osstrich/JavadocPublisher.java

```diff
@@ -147,7 +153,25 @@ private void writeIndexFiles(String groupId, Multimap<String, Artifact> artifact
 
       File indexHtml = new File(directory + "/" + majorVersion + "/index.html");
       Files.write(html, indexHtml, UTF_8);
-      gitAdd(indexHtml);
+      //gitAdd(indexHtml);
+    }
+  }
+
+  private String findRelativePath(File directory, String artifactId) {
+    if (!directory.isDirectory()) {
+      throw new IllegalArgumentException(directory + " is not a directory!");
+    }
+    if (new File(directory, "index.html").exists()) {
+      return directory.getName();
+    } else {
+      // Look for subdirectories of the same name as the artifact and go a level deeper
+      File[] subfiles = Objects.requireNonNull(directory.listFiles());
+      for (File subfile : subfiles) {
+        if (subfile.isDirectory() && artifactId.equals(subfile.getName())) {
```

Comment:
you can use the same technique for this one . . . `new File(directory, artifactId).isDirectory())`

Reply:
ah yep - 56d8fcb and 8cf31f6